tests/goth_tests/test_long_expiration/requestor.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+import asyncio
+from typing import AsyncIterable
+from datetime import timedelta
+
+from yapapi import Golem, Task, WorkContext
+from yapapi.log import enable_default_logger, log_event_repr
+from yapapi.payload import vm
+
+
+async def worker(context: WorkContext, tasks: AsyncIterable[Task]):
+    async for task in tasks:
+        script = context.new_script()
+        future_result = script.run("/bin/sh", "-c", "sleep 1000")
+
+        yield script
+
+        task.accept_result(result=await future_result)
+
+
+async def main():
+    package = await vm.repo(
+        image_hash="d646d7b93083d817846c2ae5c62c72ca0507782385a2e29291a3d376",
+    )
+
+    tasks = [Task(data=None)]
+    timeout = timedelta(hours=9)
```

Comment:
9 hours is too short as providers accept up to 10 hours by default, without mid-agreement-payments enabled... .